File: tests/resources/test_codegen/template.py
#!/usr/bin/env python3
import sys


def solve(${formal_arguments}):
    return


def main():
    def iterate_tokens():
        for line in sys.stdin:
            for word in line.split():
                yield word
    tokens = iterate_tokens()
    ${input_part}
    solve(${actual_arguments})

if __name__ == '__main__':
    main()

File: tests/resources/test_codegen/template.jl
#!/usr/bin/env julia

function solve(${formal_arguments})
  
end

function main()
    tokens = Channel{String}(32)
    Task() do
        for line in eachline(@static VERSION < v"0.6" ? STDIN : stdin)
            for token in split(chomp(line))
                put!(tokens, token)
            end
        end
        close(tokens)
    end |> schedule
    ${input_part}
    solve(${actual_arguments})
end

isempty(ARGS) && main()
